flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/exceptionhistory/ExceptionHistoryEntryMatcher.java

```diff
@@ -87,6 +107,18 @@ protected boolean matchesSafely(
             match = false;
         }
 
+        if (exceptionHistoryEntry.getLabels() == null) {
+            if (expectedLabels != null) {
+                description.appendText(" actualLabels=null");
+                match = false;
+            }
+        } else if (exceptionHistoryEntry.getLabels().equals(expectedLabels)) {
+            description
+                    .appendText(" actualLabels=")
+                    .appendText(String.valueOf(exceptionHistoryEntry.getLabels()));
+            match = false;
```

Comment:
I'm not able to wrap my head around this. How come we don't have a match if the actual labels are equal to what is expected?

Reply:
Agreed, this is now simplified